controllers/common.go

```diff
@@ -152,10 +152,11 @@ func Now() *metav1.Time {
 // * [CreateTrainingJob#TrainingJobName](https://docs.aws.amazon.com/sagemaker/latest/dg/API_CreateTrainingJob.html#SageMaker-CreateTrainingJob-request-TrainingJobName)
 // * [CreateTransformJob#TransformJobName](https://docs.aws.amazon.com/sagemaker/latest/dg/API_CreateTransformJob.html#SageMaker-CreateTransformJob-request-TransformJobName)
 // * [CreateHyperParameterTuningJob#HyperParameterTuningJobName](https://docs.aws.amazon.com/sagemaker/latest/dg/API_CreateHyperParameterTuningJob.html#SageMaker-CreateHyperParameterTuningJob-request-HyperParameterTuningJobName)
-func GetGeneratedJobName(objectMetaUID types.UID, objectMetaName string, maxNameLen int) string {
-	requiredPostfix := strings.Replace(string(objectMetaUID), "-", "", -1)
-
+func GetGeneratedResourceName(objectMetaUID types.UID, objectMetaName string, maxNameLen int, generatedPostfix ...string) string {
+	uid := strings.Replace(string(objectMetaUID), "-", "", -1)
+	requiredPostfix := strings.Join(generatedPostfix, "") + uid
```

Comment:
Could we instead join the requiredPostfix with a delimiter before the uid so that when we use the generatedPostfix we don't need to remember the hyphen?

For example:
`requiredPostfix := strings.Join(generatedPostfix, "") + delimiter + uid`

Reply:
> Could we instead join the requiredPostfix with a delimiter before the uid so that when we use the generatedPostfix we don't need to remember the hyphen?
> 
> For example:
> `requiredPostfix := strings.Join(generatedPostfix, "") + delimiter + uid`

I did not do this since this would require a conditional check to add the delimiter only when the generatedPostfix does exist.

Reply:
> We need to be careful of changing the format of these names. Because our installers set the image pull policy to `Always`, we risk breaking a customer setup when we change them. Let's avoid changes here unless we want to have a live-migration code.

The formats of the names are not changing here. The generated names would be the exact same as before. Can you elaborate?

Reply:
> > We need to be careful of changing the format of these names. Because our installers set the image pull policy to Always, we risk breaking a customer setup when we change them. Let's avoid changes here unless we want to have a live-migration code.
> 
> The formats of the names are not changing here. The generated names would be the exact same as before. Can you elaborate?

I am saying that although Nick's suggestion is a good one, we shouldn't break these names until v2 unless we really need to.

Reply:
Any changes to these formats would necessitate a major version upgrade.